app/src/main/java/com/github/sdpcoachme/groupevent/GroupEventDetailsActivity.kt

```diff
@@ -0,0 +1,595 @@
+                Text(
+                    modifier = Modifier.testTag(EVENT_SPORT),
+                    text = groupEvent.event.sport.sportName.uppercase(),
+                    style = MaterialTheme.typography.overline.copy(fontSize = 12.sp),
+                    color = Color.Gray
+                )
+            }
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.End
+            ) {
+                // TODO : add weather info
+                Spacer(modifier = Modifier.width(10.dp))
+                DayBox(eventStart.dayOfMonth, eventStart.month)
+            }
+            Spacer(modifier = Modifier.height(20.dp))
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.Start
+            ) {
+                Image(
+                    painter = painterResource(id = R.drawable.ic_launcher_background),
+                    contentDescription = "Organizer profile picture",
+                    modifier = Modifier
+                        .size(40.dp)
+                        .clip(CircleShape)
+                        .border(2.dp, Color.Gray, CircleShape)
+                        .padding(0.dp, 0.dp, 0.dp, 0.dp)
+                )
+                Text(
+                    modifier = Modifier.padding(10.dp, 0.dp, 0.dp, 0.dp),
+                    text = "Organized by " ,
+                    style = MaterialTheme.typography.body1,
+                    color = Color.Gray
+                )
+                ClickableText(
+                    modifier = Modifier.testTag(ORGANIZER_NAME),
+                    text = AnnotatedString("${organizer.firstName} ${organizer.lastName}"),
+                    style = MaterialTheme.typography.body1,
+                    onClick = {
+                        // Open organizer profile
+                        // If the organizer is the current user, open the profile activity, but not in edit mode
+                        // TODO : temporary until we update the way intents are handled in profile activity
+                        val intent = Intent(context, ProfileActivity::class.java)
+                        intent.putExtra("isViewingCoach", true)
+                        intent.putExtra("email", organizer.email)
+                        context.startActivity(intent)
+                    }
+                )
+            }
+            Spacer(modifier = Modifier.height(20.dp))
+            IconTextRow(
+                icon = Icons.Default.Place,
+                contentDescription = "Location",
+                text = groupEvent.event.address.name,
+                tag = EVENT_LOCATION,
+                onClick = {
+                    // Open google maps with the location of the event
```

Comment:
Can we modularize this part as it appears also in `ProfileActivity`?

Reply:
Yes, created an issue: #153